repondeur/zam_repondeur/__init__.py

```diff
@@ -116,3 +120,46 @@ def setup_auth(config: Configurator, settings: dict) -> None:
     # Add routes for login and logout
     config.add_route("login", "/identification")
     config.add_route("logout", "/deconnexion")
+
+
+def get_user(request: Request) -> Optional[User]:
+    """
+    Find the user in the database based on the ID in the auth cookie
+    """
+    user_id = request.unauthenticated_userid
+    if user_id is not None:
+        user: Optional[User] = DBSession.query(User).get(user_id)
```

Comment:
Don't we want to be a bit more defensive here? Like after a DB reset the `user_id` might not exist anymore?

Reply:
In that case, `get` will return `None`, and the user will not be authenticated.